### PR TITLE
DM-30983: Separated the handling of the commands in a separate model …

### PR DESCRIPTION
**Pull request by wvreeven**
…class.

### CODE REVIEW

**Review by tribeiro — APPROVED** [review]
Looks good... I made a few comments inline, nothing major...

File: python/lsst/ts/dream/__init__.py [hunk 1]
```diff
@@ -16,11 +16,17 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
 
-try:
-    from .version import *
-except ModuleNotFoundError:
+import typing
+
+if typing.TYPE_CHECKING:
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/dream/__init__.py:21]: See
https://confluence.lsstcorp.org/display/LTS/Enabling+Mypy+in+Pytest
I'll add a comment to the code.

File: python/lsst/ts/dream/dream_csc.py [hunk 2]
```diff
@@ -91,75 +82,11 @@ async def connect(self) -> None:
...
-    async def _read_loop(self) -> None:
-        """Execute a loop that reads incoming data from the SocketServer."""
-        try:
-            while True:
-                data = await self.read()
-                self.log.info(f"Received data {data!r}")
-        except Exception:
-            self.log.exception("_read_loop failed")
+            port = self.mock_port
+        await self.model.connect(host=host, port=port)
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/dream/dream_csc.py:89]: The check is in line 82.

File: python/lsst/ts/dream/__init__.py [hunk 3]
```diff
@@ -16,11 +16,17 @@
...
+else:
+    try:
+        from .version import *
+    except ImportError:
+        __version__ = "?"
 
 from .config_schema import CONFIG_SCHEMA
 from .dream_csc import *
 from .mock import *
+from .model import *
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/dream/__init__.py:32]: Thanks for the advice! I removed the import.

File: python/lsst/ts/dream/model/dream_model.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    sent_commands: `List[int]`
+        A list of command IDs that have been sent but not have been replied to
+        yet.
+    received_cmd_ids: `List[int]`
+        A list of command IDs for which a reply has been received. This is used
+        in unit tests.
+    """
+
+    def __init__(self, log: logging.Logger) -> None:
+        self.log = log.getChild(type(self).__name__)
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/dream/model/dream_model.py:58]: Done

**Pull request merged**